# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsRange.h
# Commit: 260574c2a78e
# Full Hash: 260574c2a78e619e6c9815a797a8df7086004994
# Author: Serban Stanca <sstanca@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Revert "Bug 1932150 - Update ranges to correctly set and update nodes for flattened selection r=jjaschke,smaug,dom-core"
#   
#   This reverts commit 0221ba097eb5884f3efe76c607cb0cc7c9170635.
# ==============================================================================

diff -r d699f1b53a1e -r 260574c2a78e dom/base/nsRange.h
--- a/dom/base/nsRange.h	Thu May 08 22:53:05 2025 +0300
+++ b/dom/base/nsRange.h	Thu May 08 22:53:06 2025 +0300
@@ -100,8 +100,7 @@
   static already_AddRefed<nsRange> Create(
       const mozilla::RangeBoundaryBase<SPT, SRT>& aStartBoundary,
       const mozilla::RangeBoundaryBase<EPT, ERT>& aEndBoundary,
-      ErrorResult& aRv,
-      AllowRangeCrossShadowBoundary = AllowRangeCrossShadowBoundary::No);
+      ErrorResult& aRv);
 
   NS_DECL_ISUPPORTS_INHERITED
   NS_IMETHODIMP_(void) DeleteCycleCollectable(void) override;
@@ -161,23 +160,17 @@
    * collapsed at the end point.  Similarly, if they are in different root,
    * the range will be collapsed at the end point.
    */
-  nsresult SetStartAndEnd(
-      nsINode* aStartContainer, uint32_t aStartOffset, nsINode* aEndContainer,
-      uint32_t aEndOffset,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
-          AllowRangeCrossShadowBoundary::No) {
+  nsresult SetStartAndEnd(nsINode* aStartContainer, uint32_t aStartOffset,
+                          nsINode* aEndContainer, uint32_t aEndOffset) {
     return SetStartAndEnd(RawRangeBoundary(aStartContainer, aStartOffset),
-                          RawRangeBoundary(aEndContainer, aEndOffset),
-                          aAllowCrossShadowBoundary);
+                          RawRangeBoundary(aEndContainer, aEndOffset));
   }
   template <typename SPT, typename SRT, typename EPT, typename ERT>
   nsresult SetStartAndEnd(
       const mozilla::RangeBoundaryBase<SPT, SRT>& aStartBoundary,
-      const mozilla::RangeBoundaryBase<EPT, ERT>& aEndBoundary,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
-          AllowRangeCrossShadowBoundary::No) {
-    return AbstractRange::SetStartAndEndInternal(
-        aStartBoundary, aEndBoundary, this, aAllowCrossShadowBoundary);
+      const mozilla::RangeBoundaryBase<EPT, ERT>& aEndBoundary) {
+    return AbstractRange::SetStartAndEndInternal(aStartBoundary, aEndBoundary,
+                                                 this);
   }
 
   /**
